src/setupTests.ts

```diff
@@ -77,3 +78,22 @@ beforeEach(() => {
   // Mock useWeb3React to return a chainId of 1 by default.
   mocked(useWeb3React).mockReturnValue({ chainId: 1 } as ReturnType<typeof useWeb3React>)
 })
+
+/**
+ * Fail tests if anything is logged to the console. This keeps the console clean and ensures test output stays readable.
+ * If something should log to the console, it should be stubbed and asserted:
+ * @example
+ * beforeEach(() => jest.spyOn(console, 'error').mockReturnsValue())
```

Comment:
nit: Historically, I often surrounded my code blocks with triple `, like on Github. You could consider doing that too.

Reply:
this is not required for jsdoc. See https://jsdoc.app/tags-example.html.
jsdoc examples should automatically be displayed as highlighted code.

Reply:
You are right. However, I think in context of a TypeScript file, we should refer to TSDoc instead, which is the standard used by different tools within TypeScript ecosystem (including Visual Studio Code) to work with comments. 

While JSDoc comments may work, it is not guaranteed:
> As much as possible, TSDoc should preserve the familiar style of JSDoc/Markdown. This also maximizes the likelihood that legacy comments will parse correctly as TSDoc

That said, TSDoc [uses ```  to annotate code snippets](https://tsdoc.org/pages/tags/example/). This is also a format [used by Docosaurus](https://docusaurus.io/docs/markdown-features/code-blocks), which increases interoperability, in case we ever have code snippets in other parts of our codebase that serve as documentation.